Realm/Realm.UnityWeaver/UnityWeaver.cs

```diff
@@ -467,12 +468,50 @@ private static string GetTargetOSName(RuntimePlatform target)
             };
         }
 
-        private static (string Name, string Version, string TargetArchitecture) GetNetFrameworkInfo()
+        private static (string Name, string Version) GetNetFrameworkInfo()
         {
             var targetGroup = BuildPipeline.GetBuildTargetGroup(EditorUserBuildSettings.activeBuildTarget);
             return (PlayerSettings.GetScriptingBackend(targetGroup).ToString(),
-                PlayerSettings.GetApiCompatibilityLevel(targetGroup).ToString(),
-                AnalyticsUtils.ConvertUnityArchitectureToMetricsVersion(PlayerSettings.GetArchitecture(targetGroup)));
+                PlayerSettings.GetApiCompatibilityLevel(targetGroup).ToString());
+        }
+
+        private static string GetTargetCpuArchitecture(BuildTarget buildTarget)
+        {
+            return buildTarget switch
+            {
+                BuildTarget.iOS or BuildTarget.tvOS => CpuArchitecture.Arm64,
+                BuildTarget.StandaloneOSX => EditorUserBuildSettings.GetPlatformSettings(BuildPipeline.GetBuildTargetName(buildTarget), "Architecture") switch
+                {
+                    "ARM64" => CpuArchitecture.Arm64,
+                    "x64" => CpuArchitecture.X64,
+                    _ => CpuArchitecture.Universal,
+                },
+                BuildTarget.StandaloneWindows => CpuArchitecture.X86,
+                BuildTarget.Android => PlayerSettings.Android.targetArchitectures switch
+                {
+                    AndroidArchitecture.ARMv7 => CpuArchitecture.Arm,
+                    AndroidArchitecture.ARM64 => CpuArchitecture.Arm64,
+
+                    // These two don't have enum values in our Unity reference dll, but exist in newer versions
+                    // See https://github.com/Unity-Technologies/UnityCsReference/blob/70abf502c521c169ee8a302aa48c5600fc7c39fc/Editor/Mono/PlayerSettingsAndroid.bindings.cs#L14
+                    (AndroidArchitecture)(1 << 2) => CpuArchitecture.X86,
+                    (AndroidArchitecture)(1 << 3) => CpuArchitecture.X64,
+                    _ => CpuArchitecture.Universal,
+                },
+                BuildTarget.StandaloneWindows64 or BuildTarget.StandaloneLinux64 or BuildTarget.XboxOne => CpuArchitecture.X64,
```

Comment:
How do you know and how have you tested that `BuildTarget.XboxOne` is 64 bit?

Reply:
I read on Wikipedia that Xbox One and newer consoles are 64 bit. Haven't tested it and only added it because I'm thinking we could setup the uwp wrappers to deploy on Xbox builds as experimental support.